### DISCUSSION TITLE
create new cosmos denom from JS?

### DISCUSSION THREAD

**Discussion by dckc** [Agoric/agoric-sdk#10064] [Q&A] [answered]
In the vbank, we have `uist` and `ubld`, plus incoming IBC denoms for ATOM etc.

Is there an API to create other cosmos denoms from JS? @michaelfig do you know?

**Comment by michaelfig** [comment, marked as answer]
`BankManager#addAsset` is the API you want.  That maps any Cosmos denom string to an ERTP mint or issuer.  Send some of that ERTP to a bank purse, and it will be reflected as the designated denom in the corresponding address's `x/bank` account.
